website/content/tutorials/4-constraints-matches/_index.md

```diff
@@ -0,0 +1,591 @@
+                    <formal-name>Vendor Website</formal-name>
+                    <description>A public website made by the vendor documenting their parts as used in the computer.</description>
+                  </define-field>
+                </model>
+              </define-assembly>
+              <define-field name="product-name" as-type="string" min-occurs="1" max-occurs="1">
+                <formal-name>Product Name</formal-name>
+                <description>The product name from the vendor of the computer part.</description>
+              </define-field>
+              <define-field name="type" as-type="string" min-occurs="1" max-occurs="1">
+                <formal-name>Expansion Card Type</formal-name>
+                <description>The type of expansion card on a motherboard of a computer, such as <code>pci</code> (PCI, e.g. Peripheral Component Interconnect), <code>pcie</code> (PCI Express), or an alternative.</description>
+              </define-field>
+            </model>
+          </define-assembly>
+        </model>
+        <constraint>
+          <expect id="memory-same-byte-size" level="CRITICAL" target="." test="if (count(./memory/byte-size) > 0) then (sum(./memory/byte-size) mod ./memory/byte-size[1]) = 0 else (sum(./memory/size) mod ./memory/size[1]) = 0">
+              <message>All memory modules SHOULD be the same size or byte-size for a computer.</message>
+          </expect>
+        </constraint>
+      </define-assembly>
+    </model>
+  </define-assembly>
+</METASCHEMA>
+```
+
+## Type subsets with matches constraints
+
+After previous success with complex constraints, our stakeholders want to propose a more ambitious change to the consortium. The stakeholders want flexible metadata for computer document instances, but also want precise validation for certain consortium or vendor-specific-metadata on a case-by-case basis. They approach us with a specific example. Prior to the use of our modeling tools, the stakeholders used a document management system with a flexible key-value metadata structure. Different records had additional fields, but despite that every record always had the two fields below.
+
+| Key             | Value Description | Valid Value | Invalid Value |
+|-----------------|-------------------|---------------------|-----------------------|
+|  release-status | A description of the current release status for this model of computer: "unknown", "not-applicable", "in-development", "public", "obsolete" | `public` | `READY` |
+|  release-date   | A date with year month day and timezone for when the last release status information was updated | `2024-01-01-04:00` |  `January 1` |
+
+The stakeholders want to know if we can enhance our models can define and enforce these requirements for specifically these key-value pairs, but flexibly permit others? Fortunately for us, we know we can achieve this `matches` constraints.
```

Comment:
```suggestion
Our stakeholders want to know if we can enhance our models to define and enforce these requirements for specific key-value pairs, but flexibly permit others? Fortunately for us, we know we can achieve this `matches` constraints.
```

Reply:
Edited in https://github.com/usnistgov/metaschema/pull/539/commits/d86f1787554fa86006d42eaf1630d0b0e7ce49b0.